### PR TITLE
Roles states 

### PR DESCRIPTION
**Pull request by DMinnai**

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

File: src/state/roles/types.js [hunk 1]
```diff
@@ -1,2 +1,2 @@
 // eslint-disable-next-line
-export const ADD_ROLES = 'api/fetch-roles';
+export const SET_ROLES = 'api/set-roles';
```
Comment (alessioscalici) [inline src/state/roles/types.js]: better prefix the action with "roles": 'api/set-roles' -> 'roles/set-roles'

File: test/api/roles.test.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    jest.clearAllMocks();
+  });
+
+  describe('getRoles', () => {
+    it('returns a promise', () => {
+      expect(getRoles()).toBeInstanceOf(Promise);
+    });
+
+    it('get role page 1 by default', () => {
+      getRoles({ page: 1, pageSize: 10 });
```
Comment (alessioscalici) [inline test/api/roles.test.js]: if you are testing the default, you should not pass any argument

File: test/state/roles/actions.test.js [hunk 3]
```diff
@@ -1,36 +1,105 @@
...
+
+  describe('fetchRoles', () => {
+    it('fetchRoles calls setRoles and setPage actions', (done) => {
+      store.dispatch(fetchRoles()).then(() => {
+        const actions = store.getActions();
+        expect(actions).toHaveLength(2);
+        expect(actions[0].type).toEqual(SET_ROLES);
+        expect(actions[1].type).toEqual(SET_PAGE);
+        done();
+      });
```
Comment (alessioscalici) [inline test/state/roles/actions.test.js]: add catch: `}).catch(done.fail)`

File: test/state/roles/actions.test.js [hunk 4]
```diff
@@ -1,36 +1,105 @@
...
+
+    it('role is defined and properly valued', (done) => {
+      store.dispatch(fetchRoles()).then(() => {
+        const actionPayload = store.getActions()[0].payload;
+        expect(actionPayload.roles).toHaveLength(10);
+        const role = actionPayload.roles[0];
+        expect(role).toHaveProperty('code', 'contentEditing');
+        expect(role).toHaveProperty('name');
+        done();
+      });
```
Comment (alessioscalici) [inline test/state/roles/actions.test.js]: add catch: `}).catch(done.fail)`

File: test/state/roles/actions.test.js [hunk 5]
```diff
@@ -1,36 +1,105 @@
...
-    const mockStore = configureMockStore([thunk]);
-    const store = mockStore({});
+    it('roles page two is retrieved correctly and properly valued', (done) => {
+      store.dispatch(fetchRoles({ page: 2, pageSize: 2 })).then(() => {
+        const actionPayload = store.getActions()[0].payload;
+        expect(actionPayload.roles).toHaveLength(2);
+        expect(actionPayload.roles[0]).toHaveProperty('code', 'ratingEditing');
+        expect(actionPayload.roles[1]).toHaveProperty('code', 'manageWebDynamicForms');
+        done();
+      });
```
Comment (alessioscalici) [inline test/state/roles/actions.test.js]: add catch: `}).catch(done.fail)`

File: test/state/roles/actions.test.js [hunk 6]
```diff
@@ -1,36 +1,105 @@
...
-    it('calls addGroups action', (done) => {
+    it('page is defined and properly valued', (done) => {
       store.dispatch(fetchRoles()).then(() => {
+        const actionPayload = store.getActions()[1].payload.page;
+        expect(actionPayload).toHaveProperty('page', 1);
+        expect(actionPayload).toHaveProperty('pageSize', 10);
+        expect(actionPayload).toHaveProperty('lastPage', 1);
+        expect(actionPayload).toHaveProperty('totalItems', 10);
+        done();
+      });
```
Comment (alessioscalici) [inline test/state/roles/actions.test.js]: add catch: `}).catch(done.fail)`

File: test/state/roles/actions.test.js [hunk 7]
```diff
@@ -1,36 +1,105 @@
...
+
+    it('page 2 is defined and properly valued', (done) => {
+      store.dispatch(fetchRoles({ page: 2, pageSize: 2 })).then(() => {
+        const actionPayload = store.getActions()[1].payload.page;
+        expect(actionPayload).toHaveProperty('page', 2);
+        expect(actionPayload).toHaveProperty('pageSize', 2);
+        expect(actionPayload).toHaveProperty('lastPage', 5);
+        expect(actionPayload).toHaveProperty('totalItems', 10);
+        done();
+      });
```
Comment (alessioscalici) [inline test/state/roles/actions.test.js]: add catch: `}).catch(done.fail)`

File: test/state/roles/actions.test.js [hunk 8]
```diff
@@ -1,36 +1,105 @@
...
+    it('when postRole succeeds, should dispatch SET_ROLES', (done) => {
+      store.dispatch(sendPostRole(BODY_OK)).then(() => {
         const actions = store.getActions();
-        expect(actions[0].type).toEqual(ADD_ROLES);
+        expect(actions).toHaveLength(1);
+        expect(actions[0]).toHaveProperty('type', SET_ROLES);
+        // insert when available
+        // expect(gotoRoute).toHaveBeenCalled();
         done();
       });
```
Comment (alessioscalici) [inline test/state/roles/actions.test.js]: add catch: `}).catch(done.fail)`

**Pull request merged**